datafusion/functions-array/src/lib.rs

```diff
@@ -90,5 +92,19 @@ pub fn register_all(registry: &mut dyn FunctionRegistry) -> Result<()> {
         }
         Ok(()) as Result<()>
     })?;
+
+    let rules: Vec<(AnalyzerRuleRef, usize)> = vec![
+        // The order here should be adjusted based on the rules defined in the optimizer
```

Comment:
This seems quite brittle as it would potentially break if either the users introduced their own analyzer rules or we introduced new ones in DataFusion. I think the `RewriteRule` API might avoid this potential problem.